internal/pkg/describe/service.go

```diff
@@ -358,3 +255,106 @@ func formatAppRunnerUrl(serviceURL string) string {
 
 	return svcUrl.String()
 }
+
+// newServiceStackDescriber instantiates the core elements of a new service.
+func newServiceStackDescriber(opt NewServiceConfig) (*serviceStackDescriber, error) {
+	environment, err := opt.ConfigStore.GetEnvironment(opt.App, opt.Env)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %s: %w", opt.Env, err)
+	}
+	sess, err := sessions.NewProvider().FromRole(environment.ManagerRoleARN, environment.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &serviceStackDescriber{
+		app:     opt.App,
+		service: opt.Svc,
+		env:     opt.Env,
+
+		cfn:  stack.NewStackDescriber(cfnstack.NameForService(opt.App, opt.Env, opt.Svc), sess),
+		sess: sess,
+	}, nil
+}
+
+// envVar contains serialized environment variables for a service.
+type envVar struct {
+	Environment string `json:"environment"`
+	Name        string `json:"name"`
+	Value       string `json:"value"`
+}
+
+type envVars []*envVar
+
+func (e envVars) humanString(w io.Writer) {
+	headers := []string{"Name", "Environment", "Value"}
+	var rows [][]string
+	sort.SliceStable(e, func(i, j int) bool { return e[i].Environment < e[j].Environment })
+	sort.SliceStable(e, func(i, j int) bool { return e[i].Name < e[j].Name })
+
+	for _, v := range e {
+		rows = append(rows, []string{v.Name, v.Environment, v.Value})
+	}
+
+	printTable(w, headers, rows)
+}
+
+type containerEnvVar struct {
+	*envVar
+
+	Container string `json:"container"`
+}
+
+type containerEnvVars []*containerEnvVar
+
+func (e containerEnvVars) humanString(w io.Writer) {
+	headers := []string{"Name", "Container", "Environment", "Value"}
+	var rows [][]string
+	sort.SliceStable(e, func(i, j int) bool { return e[i].Environment < e[j].Environment })
+	sort.SliceStable(e, func(i, j int) bool { return e[i].Container < e[j].Container })
+	sort.SliceStable(e, func(i, j int) bool { return e[i].Name < e[j].Name })
+
+	for _, v := range e {
+		rows = append(rows, []string{v.Name, v.Container, v.Environment, v.Value})
+	}
+
+	printTable(w, headers, rows)
+}
+
+// ServiceConfig contains serialized configuration parameters for a service.
+type ServiceConfig struct {
+	Environment string `json:"environment"`
+	Port        string `json:"port"`
+	CPU         string `json:"cpu"`
+	Memory      string `json:"memory"`
+	Platform    string `json:"platform,omitempty"`
+}
+
+type ECSServiceConfig struct {
```

Comment:
Should we move these two and their methods to the top of the file since they are exported?

Reply:
I moved them above env vars but below describers!